protocols/gateway/handle.go

```diff
@@ -0,0 +1,55 @@
+package gateway
+
+import (
+	"errors"
+	"fmt"
+
+	goHttp "net/http"
+
+	http "github.com/taubyte/http"
+	"github.com/taubyte/p2p/streams/client"
+	tunnel "github.com/taubyte/p2p/streams/tunnels/http"
+)
+
+func (g *Gateway) attach() {
+	g.http.LowLevel(&http.LowLevelDefinition{
+		PathPrefix: "/",
+		Handler: func(w goHttp.ResponseWriter, r *goHttp.Request) {
+			if err := g.handleHttp(w, r); err != nil {
+				w.Write([]byte(err.Error()))
+				w.WriteHeader(500)
+			}
+		},
+	})
+}
+
+func (g *Gateway) handleHttp(w goHttp.ResponseWriter, r *goHttp.Request) error {
+	resCh, err := g.substrateClient.ProxyHTTP(r.Host, r.URL.Path, r.Method)
+	if err != nil {
+		return fmt.Errorf("substrate client proxyHttp failed with: %w", err)
+	}
+
+	highScore := -1
```

Comment:
var match *client.Response
	clearMode := false
	for  response := range resCh {
		if clearMode {
		   defer response.Close()
		   continue
		}
		if match == nil {
		   match = response
		}  else if g.Get(response).Cached() {
		        defer match.Close()
			match = response
                        clearMode = true
		} 	 
	}
	if match == nil {
		return errors.New("no substrate match found")
	}
	defer match.Close()